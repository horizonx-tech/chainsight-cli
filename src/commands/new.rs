use clap::Parser;

#[derive(Debug, Parser)]
#[command(name = "new")]
/// Generate ChainSight project by prepared template
pub struct NewOpts {}