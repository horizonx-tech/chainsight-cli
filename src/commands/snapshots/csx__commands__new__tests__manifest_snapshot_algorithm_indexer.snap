---
source: src/commands/new.rs
expression: template_algorithm_indexer_manifest(COMPONENT_PREFIX).to_str_as_yaml().unwrap()
---
# yaml-language-server: $schema=https://raw.githubusercontent.com/horizonx-tech/chainsight-cli/main/resources/schema/algorithm_indexer.json
version: v1
metadata:
  label: sample_algorithm_indexer
  type: algorithm_indexer
  description: ''
  tags:
  - Ethereum
  - Account
datasource:
  principal: be2us-64aaa-aaaaa-qaabq-cai
  input:
    name: Transfer
    fields:
      from: String
      to: String
      value: String
  from: 17660942
  method: proxy_call
  source_type: event_indexer
output:
- name: Account
  fields:
    address: String
  output_type: key_value
interval: 3600

